utils/src/com/cloud/utils/Profiler.java

```diff
@@ -23,24 +23,20 @@
     private Long startTickInMs;
     private Long stopTickInMs;
 
-    public Profiler() {
-        startTickInMs = null;
-        stopTickInMs = null;
-    }
-
     public long start() {
-        startTickInMs = System.currentTimeMillis();
+        startTickInMs = System.nanoTime();
         return startTickInMs.longValue();
     }
 
     public long stop() {
-        stopTickInMs = System.currentTimeMillis();
+        stopTickInMs = System.nanoTime();
         return stopTickInMs.longValue();
     }
 
     public long getDuration() {
-        if (startTickInMs != null && stopTickInMs != null)
+        if (startTickInMs != null && stopTickInMs != null) {
             return stopTickInMs.longValue() - startTickInMs.longValue();
```

Comment:
I did... and it doesn't change. Only the resolution does,

Quick googled that one here for you: https://blogs.oracle.com/dholmes/entry/inside_the_hotspot_vm_clocks

Implementation is still the same:

![image](https://cloud.githubusercontent.com/assets/5129209/8301764/54ac5202-1991-11e5-9522-2567af5454b7.png)

http://stackoverflow.com/questions/351565/system-currenttimemillis-vs-system-nanotime

Cheers,
Wilder